File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {

    private int[] preSum;

    public NumArray(int[] nums) {
        preSum=new int[nums.length+1];

        for(int i=1; i<preSum.length; i++){
            preSum[i]=preSum[i-1]+nums[i-1];
        }   
    }
    
    public int sumRange(int left, int right) {
        return preSum[right+1]-preSum[left];
    }
    
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.kt
class NumArray(val nums: IntArray) {

    fun sumRange(i: Int, j: Int): Int {
        var sum = 0
        for(index in i..j) {
            sum+= nums[index]
        }
        return sum
    }

}

/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */